### ISSUE TITLE
Python 2 compatibility

### ISSUE THREAD

**Issue by hazmat345** [loganasherjones/yapconf#34]
* Yapcon version: 0.2.1
* Python version: 2.7.14
* Operating System: Centos 6

Ran into an exception while attempting to migrate a config file in Python 2 - looks like the `open` builtin didn't get the `encoding` argument until Python 3.

Stack Trace:
```
  File "/opt/beer-garden/lib/python2.7/site-packages/yapconf/spec.py", line 272, in migrate_config_file
    output_file_type)
  File "/opt/beer-garden/lib/python2.7/site-packages/yapconf/spec.py", line 371, in _write_dict_to_file
    with open(filename, 'w', encoding=self._encoding) as conf_file:
TypeError: 'encoding' is an invalid keyword argument for this function
```

**Comment by loganasherjones** [comment]
Solved by #35

**Comment by loganasherjones** [comment]
Fixed in 0.2.2
